src/mlpack/methods/ann/layer/bilinear_interpolation.hpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file bilinear_interpolation.hpp
+ * @author Kris Singh and Shikhar Jaiswal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+#define MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Definition and Implementation of the Bilinear Interpolation Layer.
+ *
+ * Bilinear Interpolation is an mathematical technique, primarily used for
+ * image scaling purposes. It is an extension of linear interpolation, for
+ * interpolating functions of two variables on the image grid. The key idea
+ * is to perform linear interpolation first in one direction (e.g., along
+ * x-axis), and then again in the other direction (i.e., y-axis), on four
+ * different known pixel points in a rectangular grid. In this way, we
+ * represent any arbitrary pixel, present within the rectangular grid,
+ * as a function of those four pixels.
```

Comment:
I suggest to remove the paragraph above since anyone knows what Bilinear Interpolation is.

Reply:
Please refer [here](https://github.com/mlpack/mlpack/pull/1238#discussion_r167863833).

Reply:
Ah, I see. I just don't like the comment since it is focused on images only. But Bilinear Interpolation is a much more general technique. Could you replace `images` by `functions` and `pixels` by `points`?